src/games/strategy/engine/framework/ui/NewGameChooserEntry.java

```diff
@@ -42,7 +44,13 @@ public NewGameChooserEntry(final URI uri)
       throws IOException, GameParseException, SAXException, EngineVersionException {
     m_url = uri;
     final AtomicReference<String> gameName = new AtomicReference<>();
-    try (InputStream input = uri.toURL().openStream()) {
+
+    Optional<InputStream> inputStream = UrlStreams.openStream(uri);
+    if(!inputStream.isPresent()) {
+      throw new IOException("Failed to open an input stream to: " + uri);
```

Comment:
methods that are marked as `throws IOException`, we'll throw an exception explicitly now to maintain prior behavior. `UrlStreams.openStream` does not throw any checked exceptions, so we have to throw an IOException to avoid refactoring clients of this code (this method throws a checked exception, "IOException")